addons/fix-uploaded-svgs/userscript.js

```diff
@@ -1,4 +1,64 @@
 export default async function ({ addon, console }) {
+  function fixEditorSVG(element) {
+    const svg = element.cloneNode(true);
+
+    // Fixes the previous issue of SVGs imported from Affinity Designer
+    if (svg.height.baseVal.valueAsString === "100%" && svg.width.baseVal.valueAsString === "100%") {
+      svg.removeAttribute("height");
+      svg.removeAttribute("width");
+    }
+
+    if (addon.settings.get("removeUnsupported")) {
+      // "dominant-baseline" needs this iframe to shift accordingly
+      const iframe = document.createElement("iframe");
+      iframe.setAttribute("src", "about:blank");
+      document.body.append(iframe);
+      iframe.contentDocument.body.appendChild(svg);
+
+      for (const textElement of svg.getElementsByTagName("text")) {
+        // "x" and "y" aren't accounted for by Scratch
+        let x = Number.parseFloat(textElement.getAttribute("x") ?? "0");
+        let y = Number.parseFloat(textElement.getAttribute("y") ?? "0");
+
+        // The lack of "tspan" creates a vertical shift
+        const innerTextElement = textElement.firstChild;
+        if (innerTextElement.tagName === undefined) {
+          const innerTextSpan = document.createElementNS("http://www.w3.org/2000/svg", "tspan");
+          innerTextSpan.appendChild(innerTextElement);
+          textElement.appendChild(innerTextSpan);
+        }
+
+        // "dominant-baseline" adjustment
+        const baselineAdjust = textElement.cloneNode(true);
+        textElement.after(baselineAdjust);
+        baselineAdjust.style.dominantBaseline = "auto";
+        const difference = textElement.getBoundingClientRect().top - baselineAdjust.getBoundingClientRect().top;
+        textElement.style.dominantBaseline = "auto";
+        baselineAdjust.remove();
+
+        let transform = textElement.getAttribute("transform") ?? "";
+        transform += ` translate(${x}, ${y + difference})`;
+        textElement.setAttribute("transform", transform);
```

Comment:
@StickmanRed Just wondering, can a transform attribute have multiple translate() rules? Is the case where the existing transform attribute already has a translate() rule considered here?

Reply:
> can a transform attribute have multiple translate() rules?

The first example snippet in [the specification for the transform attribute](https://www.w3.org/TR/SVG11/coords.html#TransformAttribute) contains two translate() functions.